src/core/Akka/Actor/Props.cs

```diff
@@ -22,8 +22,76 @@ namespace Akka.Actor
     ///   private Props otherProps = props.WithDeploy(deployment info);
     ///  </code>
     /// </summary>
-    public class Props
+    public class Props : IEquatable<Props>
     {
+        public bool Equals(Props other)
+        {
+            if (ReferenceEquals(null, other)) return false;
+            if (ReferenceEquals(this, other)) return true;
+            return CompareDeploy(other) && CompareSupervisorStrategy(other) && CompareArguments(other) && CompareInputType(other) && CompareProducer(other);
```

Comment:
Currently mocked, how should we do equality checks on Props? arguments have been serialized and can be anything, so we cant expect args to support euquality and thus, comparing the args array will be weird..